std/bitmanip.d

```diff
@@ -3272,6 +3273,106 @@ private template canSwapEndianness(T)
     }
 }
 
+/++
+    A type, that is always stored in little endian order.
+
+    This is helpful, for dealing with data structures, where numbers have to be
+    stored in little endian order.
+  +/
+struct LittleEndian(T)
+if (canSwapEndianness!T)
+{
+    version (LittleEndian)
+    {
+        private T value;
+    }
+    else
+    {
+        private T le_value;
+
+        private T value()
+        {
+            return le_value.swapEndian;
+        }
+
+        private void value(T v)
+        {
+            le_value = v.swapEndian;
+        }
+    }
+
+    alias value this;
+
+    string toString()
+    {
+        import std.conv : to;
+        return to!string(value);
+    }
+}
+
+///
+@safe pure nothrow @nogc unittest
+{
+    union Test
+    {
+        ubyte[4] a;
+        LittleEndian!ushort b;
+        LittleEndian!short c;
+        LittleEndian!uint d;
+    }
+
+    auto test = Test();
+    test.b = 3940;
+
+    version (LittleEndian)
+    {
+        assert(test.a[0]==100 && test.a[1]==15);
+    }
+    else
+    {
+        assert(test.a[0]==15 && test.a[1]==100);
+    }
+
+    test.c = -3940;
+
+    version (LittleEndian)
+    {
+        assert(test.a[0]==156 && test.a[1]==240);
+    }
+    else
+    {
+        assert(test.a[0]==240 && test.a[1]==156);
+    }
+
+    test.d = 1;
+
+    version (LittleEndian)
+    {
+        assert(test.a[0]==1 && test.a[1]==0 && test.a[2]==0 && test.a[3]==0);
+    }
+    else
+    {
+        assert(test.a[0]==0 && test.a[1]==0 && test.a[2]==0 && test.a[3]==1);
```

Comment:
Why are you testing for big endian order on big endian architectures? I thought the entire point of the type was to be the same on both.